Phase1 Pixel DQM Plot Changes

schneiml opened the pull request:
Some larger changes to the Phase1 DQM to bring the selection of plots closer to what was agreed on with the pixel experts. This reqired also implementing some missing features and fixing some bugs. (also @fioriNTU @davidcarbonis )
- Allow creating acutal `TProfile`s in step1. 
- Also allow that in 2D.
- Move larger parts of the specifications into a central place, to make it easier to change "defaults". This is pure python magic.
- Slightly change the semantics of specifications involving counts and time on one axis. This leads to a more useful output that could not be specified outherwise, but the definition is a bit vague.
- [Update] also adds the error bars in the profile-like plots from harvesting.
- [Update 2] Fix a bug in the trend plots that potentially also affected others.

schneiml committed:
Implement EXTEND_X on 1D histos

schneiml committed:
Redo Configuration, Centralize more

schneiml committed:
Implement TProfiles in step1.

schneiml committed:
Turn on profiles in config, other fixes (PerModule)

schneiml committed:
Put the quantitiy in the title, such that we don't loose it in 2D profiles.

schneiml committed:
More confguration improvements

schneiml committed:
Fix TProfile trend plots.

The Counting+Trend case is still not cleanly solved, but usable now.

cmsbuild commented:
A new Pull Request was created by  @schneiml (Marcel Schneider) for CMSSW_8_1_X.

It involves the following packages:

DQM/SiPixelPhase1Clusters
DQM/SiPixelPhase1Common
DQM/SiPixelPhase1Digis
DQM/SiPixelPhase1RecHits
DQM/SiPixelPhase1TrackClusters
DQM/SiPixelPhase1TrackEfficiency
DQM/SiPixelPhase1TrackResiduals


@cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@idebruyn, @threus, @fioriNTU, @hdelanno this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

Commit:
Error propagation added.

dmitrijus commented on DQM/SiPixelPhase1Common/src/HistogramManager.cc:
```diff
@@ -676,6 +681,7 @@ void HistogramManager::executeExtend(SummationStep& step, Table& t, bool isX) {
             // TODO Error etc.?
             new_histo.th1->SetBinContent(i, new_histo.count,
                                          th1->GetBinContent(i, j));
+	    new_histo.th1->SetBinError   (i, new_histo.count, th1->GetBinError(i, j));
```
you missed a space! ;)

schneiml replied:
I blame @davidcarbonis .

davidcarbonis replied:
Guilty as charged.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar

cmsbuild commented:
-1

Tested at: 5089d9784554f0ff518f90242a66368221d767cd

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15275/14319/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
140.53 step1
<pre>DAS Error</pre>

schneiml committed:
Fix Geometry-Values caching behaviour when step1 EXTEND is used.

This fixes the problem of UNDEFINED values appering in per-lumi trends.

schneiml committed:
Remove the alignment-based geometry code.

It is not really useful anymore, and for future reference it
will be preserved in the history.

cmsbuild commented:
Pull request #15275 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

schneiml commented:
Since the tests failed due to DAS anyways, I took the opportunity to sneak in a bugfix and a bit of cleanup.

@dmitrijus can you have a look and trigger the tests again? (once DAS is fixed maybe...)

dmitrijus commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: f7d8c486eb41459eacb014823089d3ecfc7f5bb3

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15275/14337/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
140.53 step1
<pre>DAS Error</pre>

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_1_X IBs (but tests are reportedly failing). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @smuzaffar
